core/crypto/crypto_store/crypto_suites.hpp

```diff
@@ -67,6 +68,54 @@ namespace kagome::crypto {
         gsl::span<const uint8_t> bytes) const noexcept = 0;
   };
 
+  class EcdsaSuite : public CryptoSuite<EcdsaPublicKey,
+                                        EcdsaPrivateKey,
+                                        EcdsaKeypair,
+                                        EcdsaSeed> {
+   public:
+    explicit EcdsaSuite(std::shared_ptr<EcdsaProvider> ecdsa_provider)
+        : ecdsa_provider_{std::move(ecdsa_provider)} {
+      BOOST_ASSERT(ecdsa_provider_ != nullptr);
+    }
+
+    ~EcdsaSuite() override = default;
+
+    EcdsaKeypair generateRandomKeypair() const noexcept override {
+      return ecdsa_provider_->generate().value();
```

Comment:
Why is .value() taken here without a check?

Reply:
Especially in noexcept method...